pulsar-client-api/src/main/java/org/apache/pulsar/client/internal/PulsarClientImplementationBinding.java

```diff
@@ -55,6 +55,12 @@
  * The actual implementation of this class is loaded from {@link DefaultImplementation}.
  */
 public interface PulsarClientImplementationBinding {
+    public Schema getAvroSchemaCache(Class clazz);
+    public Schema getProtobufSchemaCache(Class clazz);
+    public Schema getJsonSchemaCache(Class clazz);
+    public void setAvroSchemaCache(Class clazz,Schema schema);
+    public void setProtobufSchemaCache(Class clazz,Schema schema);
+    public void setJsonSchemaCache(Class clazz,Schema schema);
```

Comment:
The cache should be encapsulated in the PulsarClientImplementationBindingImpl and not exposed outside of the interface.